### PR TITLE
Pr/release 0.13.2 backports

### PR DESCRIPTION
**Pull request by rhuss**
## Description

Backport of critical fixes from master:

* #765 
* #774 
* #778 
* #783 
* #785 
* #788 
* #770 
* #775 
* #790 
* #799 
* #802 
* #794 
* #803 
* #797

### CODE REVIEW

**Comment by rhuss** [comment]
@googlebot I consent.

File: test/e2e/service_export_import_apply_test.go [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "List",
+		},
+		Items: []servingv1.Service{
+			getSvc(
+				withName("hello"),
+				withRevisionName("hello-rev2"),
+				withEnv("key1", "val1"),
+				withTrafficSplit([]string{"latest"}, []int{100},[]string{""}),
```
Comment (mattmoor) [inline test/e2e/service_export_import_apply_test.go:205]: Format Go code:
```suggestion
				withTrafficSplit([]string{"latest"}, []int{100}, []string{""}),
```

**Comment by navidshaikh** [comment]
@googlebot I consent.

**Comment by dsimansk** [comment]
@googlebot I consent.

**Pull request merged**